lib/husky_musher/app.py

```diff
@@ -151,121 +65,75 @@ def main():
-        # increasingly-specific labels, starting with the School.
-        "uw_school": environ.get("department", ""),
-
-        **extract_affiliation(environ),
-    }
-
-
-def extract_affiliation(environ: dict) -> Dict[str, str]:
-    """
-    Transforms a multi-value affiliation string into our REDCap fields.
-
-    Keys of the returned dict match those used by our REDCap project.
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;faculty;employee;alum"})
-    {'affiliation': 'faculty', 'affiliation_other': ''}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;student;staff"})
-    {'affiliation': 'student', 'affiliation_other': ''}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;faculty;student"})
-    {'affiliation': 'student', 'affiliation_other': ''}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;staff;alum"})
-    {'affiliation': 'staff', 'affiliation_other': ''}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;employee"})
-    {'affiliation': 'staff', 'affiliation_other': ''}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;affiliate;alum"})
-    {'affiliation': 'other', 'affiliation_other': 'affiliate;alum'}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member"})
-    {'affiliation': '', 'affiliation_other': ''}
-
-    >>> extract_affiliation({})
-    {'affiliation': '', 'affiliation_other': ''}
-    """
-    raw_affilations = environ.get("unscoped-affiliation", "")
-
-    # "Member" is uninteresting and uninformative; a generic catch-all.
-    # The empty string might arise from our fallback above.
-    affiliations = set(raw_affilations.split(";")) - {"member",""}
-
-    rules = [
-        ("student"  in affiliations,    {"affiliation": "student",  "affiliation_other": ""}),
-        ("faculty"  in affiliations,    {"affiliation": "faculty",  "affiliation_other": ""}),
-        ("staff"    in affiliations,    {"affiliation": "staff",    "affiliation_other": ""}),
-        ("employee" in affiliations,    {"affiliation": "staff",    "affiliation_other": ""}),
-        (len(affiliations) > 0,         {"affiliation": "other",    "affiliation_other": ";".join(sorted(affiliations))}),
-        (True,                          {"affiliation": "",         "affiliation_other": ""})]
-
-    return next(result for condition, result in rules if condition)
+    netid = request.form['netid'].lower().strip()
+
+    if not re.match(r'^[a-z][a-z0-9]{,7}$', netid):
+        raise InvalidNetId(netid)
+
+    redcap_record = fetch_participant({ 'netid': netid })
+
+    # Check if PT is already reigstered
+    registration_complete = redcap_registration_complete(redcap_record)
+    if not registration_complete:
+        # Give PT info on how to register
+        return render_template('registration_required.html', netid=netid,
+            redcap_record_exists=redcap_record is not None,
+            registration_complete=registration_complete)
```

Comment:
The registration_complete param could be removed here also

Reply:
Oops, thanks. I pushed a fixup to remove this from the commit it was supposed to be part of.